warehouse/forklift/legacy.py

```diff
@@ -1097,6 +1121,30 @@ def file_upload(request):
                     f"{str(version)!r}.",
                 )
 
+            filename = os.path.basename(temporary_filename)
+
+            if meta.license_files and packaging.version.Version(
+                meta.metadata_version
+            ) >= packaging.version.Version("2.4"):
+                """
+                Ensure all License-File keys exist in the wheel
+                See https://peps.python.org/pep-0639/#add-license-file-field
+                """
+                license_files = copy.deepcopy(meta.license_files)
+                with tarfile.open(temporary_filename, "r:gz") as tar:
+                    # Already validated as a tarfile by _is_valid_dist_file above
+                    member = tar.next()
+                    while member:
+                        if member.name in license_files:
+                            license_files.remove(member.name)
+                        member = tar.next()
+                if license_files != []:
```

Comment:
I'm wondering if [`getmember()`](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.getmember) might work here? Something like:

```python
with tarfile.open(temporary_filename, "r:gz") as tar:
    for license_file_path in license_files:
        try:
            tar.getmember(license_file_path)
        except KeyError:
             ...
```

This operation also has to scan the entire archive, so I don't know if this is particularly beneficial or not, similar to how you've implemented it in zipfile.

Reply:
ecda1a978502c028ec24a90fcb051b85c7222822